### PR TITLE
Implemented resource requirement module

### PR DESCRIPTION
**Pull request by dineshGramenerit**
1. Designed Resource Requirement list UI and Its functionality.
2. Implemented search option.
3. Implemented Resource Requirement list delete option.
4. Designed Resource Requirement list ADD and Edit UI.
5. Validated the ADD and EDIT forms.
6. Added the resource card for adding multiple resources.
7. Implemented the export option.

### CODE REVIEW

File: resource_management_mobile/src/app/shared/components/resource-card/resource-card.component.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+@import "../../../../theme/theme.scss";
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/shared/components/resource-card/resource-card.component.scss]: Don't import theme.scss every where declare it in angular.json or use as common global.scss

File: resource_management_mobile/src/app/pages/resource-requirement/add-resource-requirement/add-resource-requirement.component.scss [hunk 2]
```diff
@@ -0,0 +1,55 @@
+@import "../../../../theme/theme.scss";
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/pages/resource-requirement/add-resource-requirement/add-resource-requirement.component.scss]: Don't use theme.scss everywhere use it globally or declare in angular.json

File: resource_management_mobile/src/app/pages/resource-requirement/add-resource-requirement/add-resource-requirement.component.scss [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  display: flex;
+  justify-content: end;
+}
+
+.skill-add {
+  display: flex;
+  justify-content: center;
+}
+
+.add-icon {
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/pages/resource-requirement/add-resource-requirement/add-resource-requirement.component.scss]: These custom classes already have in global.scss file use them...

File: resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts [hunk 4]
```diff
@@ -1,15 +1,264 @@
...
+
+      modal.onDidDismiss().then((_) => {
+        mySubject.unsubscribe();
+      });
+    } else {
+      this.isModalOpen = false;
+    }
+  }
+
+  async deleteModal(item: any, index: number, sliding: any) {
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts]: Replace any with their datatype

File: resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts [hunk 5]
```diff
@@ -1,15 +1,264 @@
...
+          this.toastConstants.Delete_success_message
+        );
+      },
+      error: (response) => {
+        this.toastService.presentToast(this.toastConstants.try_again);
+      },
+    });
+  }
+
+  setOpen(isOpen: boolean, type: string, skillInfo?: any, index?: any) {
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts]: same here (any)

File: resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts [hunk 6]
```diff
@@ -1,15 +1,264 @@
...
+
+  async openExportModel() {
+    this.resourceRequirementService
+      .getAllResource()
+      .subscribe(async (res: ResourceResponse) => {
+        const keyToRemove = [
+          'Requirement_requirement_id',
+          'Resource_requirement_id',
+        ];
+        const newArray = res.data.resourceRequirementInfo.map((obj: any) => {
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts]: Same here (Any)

File: resource_management_mobile/src/app/pages/resource-requirement/services/resource-requirement.service.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  editResource(requestResource: editResourceRequest): Observable<editResourceRequest> {
+    return this.http.put<editResourceRequest>(`${this.URL}resource/requirement`, requestResource);
+  }
+
+  deleteResource(id: string) {
+    return this.http.delete(`${this.URL}resource/requirement/${id}`).pipe(
+      catchError((error: HttpErrorResponse) => {
+        return throwError('Error while deleting a data ' + error.message);
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/pages/resource-requirement/services/resource-requirement.service.ts]: Replace static string with enum in static.enum.ts use Common enum in this......

File: resource_management_mobile/src/app/shared/components/resource-card/resource-card.component.ts [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  selector: 'app-resource-card',
+  templateUrl: './resource-card.component.html',
+  styleUrls: ['./resource-card.component.scss'],
+  standalone: true,
+  imports: [IonicModule, CommonModule, ReactiveFormsModule],
+})
+export class ResourceCardComponent implements OnInit {
+  @Input() resourceId: any;
+  resourceform!: FormGroup;
+  resourceData: any[] = [];
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/shared/components/resource-card/resource-card.component.ts]: Replace any

File: resource_management_mobile/src/app/shared/components/resource-card/resource-card.component.scss (hunk 1 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/shared/components/resource-card/resource-card.component.scss]: Done

File: resource_management_mobile/src/app/pages/resource-requirement/add-resource-requirement/add-resource-requirement.component.scss (hunk 2 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/pages/resource-requirement/add-resource-requirement/add-resource-requirement.component.scss]: done

File: resource_management_mobile/src/app/pages/resource-requirement/add-resource-requirement/add-resource-requirement.component.scss (hunk 3 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/pages/resource-requirement/add-resource-requirement/add-resource-requirement.component.scss]: done

Comment (dineshGramenerit) [reply resource_management_mobile/src/app/pages/resource-requirement/add-resource-requirement/add-resource-requirement.component.scss]: fixed

File: resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts (hunk 4 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts]: fixed

File: resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts (hunk 5 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts]: fixed

File: resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts (hunk 6 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/pages/resource-requirement/resource-requirement.page.ts]: fixed

File: resource_management_mobile/src/app/pages/resource-requirement/services/resource-requirement.service.ts (hunk 7 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/pages/resource-requirement/services/resource-requirement.service.ts]: fixed

File: resource_management_mobile/src/app/shared/components/resource-card/resource-card.component.ts (hunk 8 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/shared/components/resource-card/resource-card.component.ts]: fixed

**Review by deviprasad0327 — APPROVED** [review]

**Pull request merged**